ray-operator/test/sampleyaml/rayjob_test.go

```diff
@@ -0,0 +1,80 @@
+package sampleyaml
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	. "github.com/ray-project/kuberay/ray-operator/test/support"
+)
+
+func TestRayJob(t *testing.T) {
+	tests := []struct {
+		name string
+	}{
+		{
+			name: "ray-job.custom-head-svc.yaml",
+		},
+		{
+			name: "ray-job.modin.yaml",
+		},
+		{
+			name: "ray-job.resources.yaml",
+		},
+		{
+			name: "ray-job.sample.yaml",
+		},
+		{
+			name: "ray-job.shutdown.yaml",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			test := With(t)
+			g := NewWithT(t)
+
+			namespace := test.NewTestNamespace()
+			test.StreamKubeRayOperatorLogs()
+			rayJobFromYaml := DeserializeRayJobSampleYAML(test, tt.name)
+			KubectlApplyYAML(test, tt.name, namespace.Name)
+
+			test.T().Logf("Waiting for RayJob %s/%s to be running", namespace.Name, rayJobFromYaml.Name)
+			g.Eventually(RayJob(test, namespace.Name, rayJobFromYaml.Name), TestTimeoutMedium).
+				Should(WithTransform(RayJobStatus, Equal(rayv1.JobStatusRunning)))
+
+			rayJob, err := GetRayJob(test, namespace.Name, rayJobFromYaml.Name)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(rayJob).NotTo(BeNil())
+
+			// Wait for RayCluster name to be populated
+			g.Eventually(RayJob(test, rayJob.Namespace, rayJob.Name), TestTimeoutShort).
+				Should(WithTransform(RayJobClusterName, Not(BeEmpty())))
+
+			rayCluster, err := GetRayCluster(test, namespace.Name, rayJob.Status.RayClusterName)
```

Comment:
You need to get RayJob again before this line, otherwise `rayJob.Status.RayClusterName` won't be updated.

```go
rayJob, err := GetRayJob(test, namespace.Name, rayJobFromYaml.Name)
g.Expect(err).NotTo(HaveOccurred())
g.Expect(rayJob).NotTo(BeNil())
```